app/helpers/security_group_helper/textual_summary.rb

```diff
@@ -70,4 +69,16 @@ def textual_cloud_tenant
   def textual_network_ports
     @record.network_ports
   end
+
+  def port_range_helper(rule)
+    if rule.host_protocol.to_s.upcase.include?("ICMP")
+      _("N/A")
```

Comment:
@tumido - According to the AWS documentation, ICMP *must* use a port range:
"For tcp, udp, and icmp, you must specify a port range.", found[ here](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_IpPermission.html)
Did I misinterpret the doc?

Reply:
Well, ports don't apply to the ICMP. It "must" be specified in the rule, though it means a message type code, not a port. From the same doc:
```
FromPort (request), fromPort (response)
.... If you specify all ICMP/ICMPv6 types, you must specify all codes.
```
That means if you want to subset the ICMP message types, you specify the exact message code you want to have there. Otherwise something like "all codes" is used which coresponds to -1 in AWS, though it might be different for other providers.

It doesn't make sense to treat message types as port numbers since nothing like port number exist for ICMP or ICMPv6.

ICMP:
https://tools.ietf.org/html/rfc792
https://en.wikipedia.org/wiki/Internet_Control_Message_Protocol

Reply:
just a bit confusing, huh? :-)

Reply:
Just a little bit /s :laughing: 
Also if you use AWS UI to define the rule the ICMP message type is selected in the column *Protocol* and the codes under column *Port range*... It's a mess just because they want to be able to map the ICMP rule on the same table as TCP/UDP ones, I suppose...